server/core/session.cc

```diff
@@ -979,9 +986,19 @@ void session_clear_stmt(MXS_SESSION *session)
     session->stmt.target = NULL;
 }
 
-uint64_t session_get_next_id()
+uint32_t session_get_next_id()
 {
-    return atomic_add_uint64(&next_session_id, 1);
+    uint32_t rval;
+    spinlock_acquire(&thread_id_lock);
+    do
+    {
+        rval = atomic_add_uint32(&next_session_id, 1);
```

Comment:
Since this is now protected by a lock, there's no need to use `atomic_add_uint32` but you can simply increment `next_session_id`.